### PR TITLE
Document constraint "s"

### PR DESCRIPTION
**Pull request by MaskRay**
"s" can be used to create an artificial reference for linker garbage collection, or define sections to hold symbol addresses. "S" does not work for a preemptible symbol in GCC while "s" does.

### CODE REVIEW

**Comment by cmuellner** [comment]
Is there a reference to GCC and/or LLVM patches?

**Review by kito-cheng — APPROVED** [review]
LGTM, RISC-V GCC already support that since day 1, and LLVM supporting also under review[1], plan to merge this PR once LLVM supporting merged :)

[1] https://github.com/llvm/llvm-project/pull/80201

**Comment by kito-cheng** [comment]
For GCC: https://github.com/gcc-mirror/gcc/blob/master/gcc/common.md#L97

**Comment by cmuellner** [comment]
I just want to mention that in GCC `s` is an arch-independent constraint: https://gcc.gnu.org/onlinedocs/gcc/Simple-Constraints.html#index-s-in-constraint

But since this won't ever change, it does not matter much.

**Comment by kito-cheng** [comment]
clang support has been merged to trunk :)

**Pull request merged**
